tests/libvmi/network.go

```diff
@@ -26,9 +26,22 @@ import (
 // WithInterface adds a Domain Device Interface.
 func WithInterface(iface kvirtv1.Interface) Option {
 	return func(vmi *kvirtv1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Devices.Interfaces = append(
-			vmi.Spec.Domain.Devices.Interfaces, iface,
-		)
+		// Replace the interface with the same binding
+		// in case there is any
+		ifaceReplaced := false
+		for idx, vmiIface := range vmi.Spec.Domain.Devices.Interfaces {
```

Comment:
wait ... can't we use the binding twice ?.... You cannot do it with masquerade (admitter only allows binding to the pod network, and you can bind to it only once), but, with Bridge (and macvtap) that is definitely possible.

Reply:
I am going to remove this, is half baked code.

Reply:
You could just drop this weird 'replace interface' section, and simply append always.

To do that, on your [createVmiCirros](https://github.com/kubevirt/kubevirt/pull/4061/files#diff-bdaa78c5363f61af93fe1376b26fd2a4R241) helper, you would need to clear the interface and network list.

Not the most elegant solution, but I find it nicer than what you're presenting currently.

Reply:
Yep that was the other option modifying the VMI and pass options.